cli/cli/commands/clean/clean.go

```diff
@@ -92,10 +83,27 @@ func run(
 			// Don't use stacktrace b/c the only reason this function exists is to pass in the right args
 			return cleanEnclaves(ctx, engineClient, shouldCleanAll)
 		},
+		unusedImagesPhaseTitle: func() ([]string, []error, error) {
+			// Don't use stacktrace b/c the only reason this function exists is to pass in the right args
+			return cleanUnusedImages(ctx, kurtosisBackend, shouldCleanAll)
+		},
+	}
+
+	executeOnlyIfAllFlagIsEnabled := map[string]bool{
+		oldEngineCleaningPhaseTitle: false,
+		enclavesCleaningPhaseTitle:  false,
+		unusedImagesPhaseTitle:      true,
 	}
 
 	phasesWithErrors := []string{}
 	for phaseTitle, cleaningFunc := range cleaningPhaseFunctions {
+		onlyIfAllFlag, found := executeOnlyIfAllFlagIsEnabled[phaseTitle]
+		if !found {
+			return stacktrace.NewError("Couldn't find phase %s in executeOnlyIfAllFlagIsEnabled. This is a bug in Kurtosis", phaseTitle)
+		}
+		if onlyIfAllFlag && !shouldCleanAll {
```

Comment:
made me think, wonder if there's a more elegant way to represent this or if i just need more coffee